libs/platform/router/src/services/default-router.service.ts

```diff
@@ -77,6 +71,8 @@ export class DefaultRouterService implements RouterService {
       return;
     }
 
+    if (this.isSameRoute(route)) return;
```

Comment:
Can we do this before checking on not found? Not only more efficient but also more logical I can imagine.